src/backend/utils/compatibility_layers.ts

```diff
@@ -548,13 +554,14 @@ export const getUmuPath = async () =>
   searchForExecutableOnPath('umu-run').then((path) => path ?? defaultUmuPath)
 
 export async function isUmuSupported(
-  wineType: WineInstallation['type'],
+  gameSettings: GameSettings,
   checkUmuInstalled = true
 ): Promise<boolean> {
-  const umuEnabled =
-    GlobalConfig.get().getSettings().experimentalFeatures?.umuSupport === true
-  const wineVersionSupported = wineType === 'proton'
-  const umuInstalled = checkUmuInstalled ? existsSync(await getUmuPath()) : true
+  if (!isLinux) return false
+  if (gameSettings.wineVersion.type !== 'proton') return false
+  if (gameSettings.disableUMU) return false
+  if (!checkUmuInstalled) return true
+  if (!existsSync(await getUmuPath())) return false
```

Comment:
I'm using individual returns for each condition cause I think it's easier to understand what's being checked and we can avoid more expensive operations if we can return earlier